src/inner-slider.js

```diff
@@ -67,32 +76,41 @@ export var InnerSlider = createReactClass({
     } else {
       window.detachEvent('onresize', this.onWindowResized);
     }
-    if (this.state.autoPlayTimer) {
-      clearInterval(this.state.autoPlayTimer);
+    if (this.autoplayTimer) {
+      clearInterval(this.autoplayTimer);
     }
   },
   componentWillReceiveProps: function (nextProps) {
-    if (this.props.slickGoTo != nextProps.slickGoTo) {
-      if (process.env.NODE_ENV !== 'production') {
-        console.warn('react-slick deprecation warning: slickGoTo prop is deprecated and it will be removed in next release. Use slickGoTo method instead')
+    let spec = assign({listRef: this.list, trackRef: this.track}, nextProps, this.state)
+    let updatedState = initializedState(spec)
+    assign(spec, {slideIndex: updatedState.currentSlide}, updatedState)
```

Comment:
Don't mutate spec. instead create a new object every time
 spec = assign({}, spec, {slideIndex: updatedState.currentSlide}, updatedState)

Reply:
I thought about it before doing so, and it didn't make much of a difference in this case. But anyway, I was thinking of using object spreading instead of using assign. I'll do that.